tests/test_buffer_manager.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import time
+from swsscommon import swsscommon
+
+class TestBufferManager(object):
+    def make_dict(self, input_list):
+        return dict(input_list[1])
+
+    def setup_db(self, dvs):
+        self.config_db = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        self.buffer_pg_table = swsscommon.Table(self.config_db, "BUFFER_PG")
+        self.port_table = swsscommon.Table(self.config_db, "PORT")
+        cable_length_table = swsscommon.Table(self.config_db, "CABLE_LENGTH")
+        cable_length_key = cable_length_table.getKeys()[0]
+        self.cable_lengths = self.make_dict(cable_length_table.get(cable_length_key))
+        self.buffer_profile_table = swsscommon.Table(self.config_db, "BUFFER_PROFILE")
+
+    def load_pg_profile_lookup(self, dvs):
+        self.profile_lookup_info = {}
+        lines = dvs.runcmd('cat /usr/share/sonic/hwsku/pg_profile_lookup.ini')[1].split('\n')
+
+        SPEED = 0
+        CABLE_LENGTH = 1
+        SIZE = 2
+        XON = 3
+        XOFF = 4
+        THRESHOLD = 5
+        XON_OFFSET = 6
+
+        for line in lines:
+            if len(line) == 0 or line[0] == '#':
+                continue
+            tokens = line.split()
+            self.profile_lookup_info[(tokens[SPEED], tokens[CABLE_LENGTH])] = {
+                'size': tokens[SIZE],
+                'xon': tokens[XON],
+                'xoff': tokens[XOFF],
+                'dynamic_th': tokens[THRESHOLD]
+            }
+            if XON_OFFSET < len(tokens):
+                self.profile_lookup_info[(tokens[SPEED], tokens[CABLE_LENGTH])]['xon_offset'] = tokens[XON_OFFSET]
+
+    def test_buffer_pg(self, dvs):
+        self.setup_db(dvs)
+
+        port = 'Ethernet0'
+        pg = port + '|3-4'
+
+        # At the beginning the port are admin down on VM. Make sure no lossless PG exists
+        assert not self.buffer_pg_table.get(pg)[0]
```

Comment:
can we check if the port is admin down and then do this check? eg. if some testcase did not cleanup properly and left this port admin up

Reply:
fixed.